src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -119,24 +119,25 @@ public class Builder : IDisposable
 
 		public string MicrosoftNetSdkDirectory {
 			get {
-				string path;
-				if (IsUnix) {
-					path = FindLatestDotNetSdk (frameworkSDKRoot);
-					if (!string.IsNullOrEmpty (path))
-						return path;
-	
-					return string.Empty;
-				}
-				var visualStudioDirectory = TestEnvironment.GetVisualStudioDirectory ();
-				if (!string.IsNullOrEmpty (visualStudioDirectory)) {
-					path = Path.Combine (visualStudioDirectory, "MSBuild", "Sdks", "Microsoft.NET.Sdk");
+				string dotnetSearchPath = IsUnix ? frameworkSDKRoot
+						:  Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.ProgramFiles), "dotnet", "sdk");
+
+				string path = FindLatestDotNetSdk (dotnetSearchPath);
+				if (!string.IsNullOrEmpty (path))
+					return path;
+
+				if (!IsUnix) {
+					var visualStudioDirectory = TestEnvironment.GetVisualStudioDirectory ();
+					if (!string.IsNullOrEmpty (visualStudioDirectory)) {
+						path = Path.Combine (visualStudioDirectory, "MSBuild", "Sdks", "Microsoft.NET.Sdk");
+						if (File.Exists (Path.Combine (path, "Sdk", "Sdk.props")))
+							return path;
+					}
+					var x86 = Environment.GetFolderPath (Environment.SpecialFolder.ProgramFilesX86);
+					path = Path.Combine (x86, "MSBuild", "Sdks", "Microsoft.NET.Sdk");
 					if (File.Exists (Path.Combine (path, "Sdk", "Sdk.props")))
 						return path;
 				}
-				var x86 = Environment.GetFolderPath (Environment.SpecialFolder.ProgramFilesX86);
-				path = Path.Combine (x86, "MSBuild", "Sdks", "Microsoft.NET.Sdk");
-				if (File.Exists (Path.Combine (path, "Sdk", "Sdk.props")))
-					return path;
```

Comment:
Hey, I was looking at this is only used in one place: https://github.com/xamarin/xamarin-android/blob/9e7b246f7e8cb46248027260e90068ff2e99efd6/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/PackagingTest.cs#L612-L613

Can we just remove that property and this assertion? I don't see how this is testing anything...

Reply:
Heh I didn't actually look at usage, I agree we should be able to remove this and just end up failing later if we don't have the right SDK installed.